kfs-core/src/main/java/edu/arizona/kfs/fp/document/web/struts/DisbursementVoucherAction.java

```diff
@@ -0,0 +1,68 @@
+package edu.arizona.kfs.fp.document.web.struts;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.struts.action.ActionForm;
+import org.apache.struts.action.ActionForward;
+import org.apache.struts.action.ActionMapping;
+import org.kuali.kfs.fp.document.service.DisbursementVoucherTaxService;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.arizona.kfs.fp.businessobject.DisbursementVoucherIncomeType;
+import edu.arizona.kfs.fp.document.DisbursementVoucherDocument;
+import edu.arizona.kfs.tax.document.IncomeTypeContainer;
+
+public class DisbursementVoucherAction extends org.kuali.kfs.fp.document.web.struts.DisbursementVoucherAction {
+
+    public ActionForward newIncomeType(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        getIncomeTypeContainer(form).getIncomeTypeHandler().addNewIncomeType();
+        return mapping.findForward(KFSConstants.MAPPING_BASIC);
+    }
+
+    public ActionForward deleteIncomeType(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        getIncomeTypeContainer(form).getIncomeTypeHandler().removeIncomeType(getLineToDelete(request));
+        return mapping.findForward(KFSConstants.MAPPING_BASIC);
+    }
+    
+    
+    private IncomeTypeContainer <DisbursementVoucherIncomeType, String>getIncomeTypeContainer(ActionForm form) {
+       DisbursementVoucherForm dvForm = (DisbursementVoucherForm)form;
+        return (IncomeTypeContainer<DisbursementVoucherIncomeType, String>)dvForm.getDocument();
+    }
+
+    public ActionForward refreshIncomeTypesFromAccountLines(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        DisbursementVoucherForm dvForm = (DisbursementVoucherForm)form;
+        DisbursementVoucherDocument doc = (DisbursementVoucherDocument)dvForm.getDocument();
+        doc.getIncomeTypeHandler().populateIncomeTypes(doc.getSourceAccountingLines());
+        return mapping.findForward(KFSConstants.MAPPING_BASIC);
+    }
+    
+    public ActionForward generateNonResidentAlienTaxLines(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
```

Comment:
It looks as though methods generateNonResidentAlienTaxLines and clearNonResidentAlienTaxLines have been copied directly from org.kuali.kfs.fp.document.web.struts.DisbursementVoucherAction with no changes. Please see KFS 3.0 code edu.arizona.kfs.fp.document.web.struts.DisbursementVoucherAction for the override of method generateNonResidentAlienTaxLines (lines 93-130). It is not necessary to override method clearNonResidentAlienTaxLines.